site/content/entry/yuzu-progress-report-apr-2022/index.md

```diff
@@ -0,0 +1,331 @@
+
+This change doesn’t include support for GLASM at the moment, since our developers aren't too fond of having to deal with NVIDIA assembly shader code.
+Imagine being asked to fix an issue in a car engine, and the only given tools for the job are a rock and a stick.
+
+However, that was only half the battle. Proper OpenGL support for `Super Mario Sunshine` and `Super Mario Galaxy` required solving an old limitation we had with the aging API: broken Z scale inversion.
+
+Most Switch games use either OpenGL, the popular free graphics API, or NVN, the proprietary NVIDIA API exclusive to the console. 
+Arguably, NVN is much closer to OpenGL than Vulkan in how it operates.
+
+The Tegra X1 GPU on the Switch is flexible enough to allow the coordinate system to be changed at the discretion of the game developer. While most games will behave closer to what OpenGL expects, with the Z-axis facing away from the camera, `Hagi` and `Hovercraft` emulated games (which render using Vulkan and is exclusive to a tiny handful of titles on the Switch) have the coordinates inverted and the Z-axis facing into the camera, the way Vulkan games would expect to natively render.
+
+{{< imgs
+	"./coords.png| byte[]'s Z-axis diagram"
+  >}}
+
+This was not an issue if you wanted to play `Super Mario Galaxy` or `Super Mario Sunshine` in yuzu with yuzu's Vulkan backend, as the behaviour matched with what the game expected. 
+But if you tried to play using OpenGL instead, yuzu would not correctly interpret that the faces were flipped due to the Z scale inversion, and thus rendered only the back faces of objects.
+
+The solution is very simple, {{< gh-hovercard "8149" "flip the front faces" >}} when the Z-axis is inverted.
+
+{{< single-title-imgs-compare
+	"Welcome to the Shadow Realm Resort (Super Mario Sunshine)"
+	"./zbug.png"
+	"./zfix.png"
+>}}
+
+Next in line, you may have noticed that `Super Mario Sunshine` rendered with a black bar at the bottom.
+This is because the Wii and GC games natively use an aspect ratio different to the usual 16:9 we’re used to.
+Instead, the games render at a 5:3 aspect ratio. 
+`Super Mario Galaxy` informs the system to explicitly crop the screen to its native resolution of 1920x1012, but `Super Mario Sunshine` does not, so yuzu previously did not attempt to crop the game, resulting in a conspicuous black bar at the bottom of the render.
+
+{{< imgs
+	"./crop.png| Diagram of the cropping process"
+  >}}
+
+While the game proportions in `Super Mario Sunshine`, arguably, appear more correct with the black bar, that’s not how Nintendo intended the game to be played. For accuracy’s sake, byte[] interprets the game's implicit crop request, which stretches the image to match the native 1920x1080 resolution of the Switch, both {{< gh-hovercard "8150" "for Vulkan" >}} and {{< gh-hovercard "8152" "for OpenGL" >}}.
+
+{{< single-title-imgs-compare
+	"Don not adjust your set (Super Mario Sunshine)"
```

Comment:
```suggestion
	"Do not adjust your set (Super Mario Sunshine)"
```
I don't understand the relevance of this TV show to Super Mario Sunshine, but if you need to keep it in, at least fix the typo.